### ISSUE TITLE
[explicit-module-boundary-types] False positive when disable return type ?

### ISSUE THREAD

**Issue by orblazer** [typescript-eslint/typescript-eslint#2532]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-function-return-type": "off"
  }
}
```

```TS
import React from 'react'

export default function App() {
  return <h1>Hello world !</h1>
}
```

**Expected Result**

No warning

**Actual Result**

![Screenshot_20200909_213457](https://user-images.githubusercontent.com/1344792/92644918-5756e800-f2e4-11ea-980b-738692d84d80.png)

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.1.0` |
| `@typescript-eslint/parser`        | `4.1.0` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.8.1` |
| `node`                             | `14.9.0` |

**Comment by bradzacher** [comment]
> explicit-function-return-type

your screenshot shows a differnet rule...
`explicit-module-boundary-types`

**Comment by orblazer** [comment]
> > explicit-function-return-type
> 
> your screenshot shows a differnet rule...
> `explicit-module-boundary-types`

Yes I know but how I can fix that without off completely the rule ? I don't find other solution 🙁
And for me that behavior is strange.
